src/chunk_index.c

```diff
@@ -636,33 +671,70 @@ chunk_index_delete(Chunk *chunk, Oid chunk_indexrelid, bool drop_index)
 				DirectFunctionCall1(namein, CStringGetDatum(indexname)));
 
 	return chunk_index_scan_update(CHUNK_INDEX_CHUNK_ID_INDEX_NAME_IDX,
-								   scankey, 2, chunk_index_tuple_delete, &drop_index);
+								   scankey, 2, chunk_index_tuple_delete, NULL, &data);
+}
+
+void
+chunk_index_delete_by_name(const char *schema, const char *index_name, bool drop_index)
+{
+	ScanKeyData scankey[1];
+
+	ChunkIndexDeleteData data = {
+		.drop_index = drop_index,
+		.schema = schema,
+	};
+
+	ScanKeyInit(&scankey[0],
```

Comment:
What is the benefit of an index scan over a heap scan if the key is only on the second index column?  Aren't you just traversing the whole index anyway?